ipfs-7-to-8/migration/config_conv.go

```diff
@@ -0,0 +1,177 @@
+		"QmQCU2EcMqAqQPR2i9bChDtGNJchTbq5TbXJJ16u19uLTa",
+		"QmbLHAnMoJPWSCR5Zhtx6BHJX9KiKNN6tpvbUcqanj75Nb",
+		"QmcZf59bWwK5XFi76CZX8cbJ4BhTzzA3gU1ZjYZcYW3dwt",
+	}
+	smallKeyBootstrapPeers = []string{
+		"QmSoLnSGccFuZQJzRadHn95W2CrSFmZuTdDWP8HXaHca9z",
+		"QmSoLueR4xBeUbY9WZ9xGUUxunbKWcrNFTDAadQJmocnWm",
+		"QmSoLMeWqB7YGVLJN3pNLQpmmEk35v6wYtsMGLzSr5QBU3",
+		"QmSoLju6m7xTh3DuokvT3886QRYqxAzb1kShaanJgW36yx",
+		"QmSoLPppuBtQSGwKDZT2M73ULpjvfd3aZ6ha4oFGL1KrGM",
+		"QmSoLSafTMBsPKadTEgaXctDQVcqN88CNLHXMkTNwMKPnu",
+		"QmSoLV4Bbm51jM9C4gDYZQ9Cy3U6aXMJDAbzgu2fzaDs64",
+		"QmSoLer265NRgSp2LA3dPaeykiS1J6DifTC88f5uVQKNAd",
+	}
+	oldBootstrapAddrs = []string{
+		"/ip4/104.131.131.82/tcp/4001/p2p/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ",
+		"/ip4/104.236.179.241/tcp/4001/p2p/QmSoLPppuBtQSGwKDZT2M73ULpjvfd3aZ6ha4oFGL1KrGM",
+		"/ip4/128.199.219.111/tcp/4001/p2p/QmSoLSafTMBsPKadTEgaXctDQVcqN88CNLHXMkTNwMKPnu",
+		"/ip4/104.236.76.40/tcp/4001/p2p/QmSoLV4Bbm51jM9C4gDYZQ9Cy3U6aXMJDAbzgu2fzaDs64",
+		"/ip4/178.62.158.247/tcp/4001/p2p/QmSoLer265NRgSp2LA3dPaeykiS1J6DifTC88f5uVQKNAd",
+		"/ip6/2604:a880:1:20::203:d001/tcp/4001/p2p/QmSoLPppuBtQSGwKDZT2M73ULpjvfd3aZ6ha4oFGL1KrGM",
+		"/ip6/2400:6180:0:d0::151:6001/tcp/4001/p2p/QmSoLSafTMBsPKadTEgaXctDQVcqN88CNLHXMkTNwMKPnu",
+		"/ip6/2604:a880:800:10::4a:5001/tcp/4001/p2p/QmSoLV4Bbm51jM9C4gDYZQ9Cy3U6aXMJDAbzgu2fzaDs64",
+		"/ip6/2a03:b0c0:0:1010::23:1001/tcp/4001/p2p/QmSoLer265NRgSp2LA3dPaeykiS1J6DifTC88f5uVQKNAd",
+	}
+)
+
+// convFunc does an inplace conversion of the "Bootstrap"
+// configuration from one version to another
+type convFunc func([]string) []string
+
+// convertFile converts a config file from one version to another, the
+// converted config is stored in
+func convertFile(orig string, new string, convFunc convFunc) error {
+	in, err := os.Open(orig)
+	if err != nil {
+		return err
+	}
+	out, err := os.Create(new)
+	if err != nil {
+		return err
+	}
+	return convert(in, out, convFunc)
+}
+
+// convert converts the config from one version to another, returns
+// the converted config as a map[string]interface{}
+func convert(in io.Reader, out io.Writer, convFunc convFunc) error {
+	data, err := ioutil.ReadAll(in)
+	if err != nil {
+		return err
+	}
+	confMap := make(map[string]interface{})
+	if err = json.Unmarshal(data, &confMap); err != nil {
+		return err
+	}
+	bootstrapi, _ := confMap["Bootstrap"].([]interface{})
+	if bootstrapi == nil {
+		bootstrapi, _ := confMap["bootstrap"].([]interface{})
+		if bootstrapi == nil {
+			return fmt.Errorf("Bootstrap field missing or of the wrong type")
+		}
+	}
+	bootstrap := make([]string, len(bootstrapi))
+	for i := range bootstrapi {
+		bootstrap[i] = bootstrapi[i].(string)
+	}
+	res := convFunc(bootstrap)
+	confMap["Bootstrap"] = res
+	fixed, err := json.MarshalIndent(confMap, "", "  ")
+	if err != nil {
+		return err
+	}
+	out.Write(fixed)
+	out.Write([]byte("\n"))
+	return nil
+}
+
+func ver7to8(bootstrap []string) []string {
```

Comment:
If possible, let's try to avoid reordering. _Especially_ if we don't end up modifying the set of bootstrappers.